File: tests/fixtures/repomap/sample.ex
import Ecto.Query
alias App.Repo
use GenServer

defmodule App.UserService do
  def get_user(id) do
    Repo.get(User, id)
  end

  def list_users do
    Repo.all(User)
  end

  defp validate(id) do
    id > 0
  end
end

defmodule App.Config do
  def default_port, do: 8080
end

def create_app(config) do
  App.UserService.start_link(config)
end

File: tests/fixtures/repomap/sample.ml
open Printf
open Stdlib

let get_user id =
  Printf.printf "Getting user %d\n" id

let create_app config =
  Printf.printf "Creating app\n"

type user = {
  id: int;
  name: string;
}

type config = {
  port: int;
  host: string;
}

module UserService = struct
  let find id = Printf.printf "Finding %d\n" id
  let list_all () = Printf.printf "Listing all\n"
end
